src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,130 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(10);
+    }
+
+    CircularBuffer_t* circular_buffer;
+};
+
+TEST_F(CircularBufferTest, circular_buffer_exact_full)
+{
+    float mockData[]     = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0};
+    float expectedData[] = {10.0, 9.0, 8.0, 7.0, 6.0, 5.0, 4.0, 3.0, 2.0, 1.0};
+    int mockDataSize     = sizeof(mockData) / sizeof(mockData[0]);
+
+    // Populate circular buffer with mock data
+    for (int i = 0; i < mockDataSize; i++)
+    {
+        circular_buffer_push(circular_buffer, mockData[i]);
+    }
+
+    EXPECT_EQ(true, circular_buffer_isFull(circular_buffer));
+
+    for (int i = 0; i < 10; i++)
```

Comment:
nit: this `10` could also be `mock_data_size`?

Reply:
Done.